runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java

```diff
@@ -287,4 +317,50 @@ private void updateRifts()
 			rifts.add(WATER_RIFT);
 		}
 	}
+
+	@Subscribe
+	public void onGameObjectSpawned(GameObjectSpawned event)
+	{
+		GameObject obj = event.getGameObject();
+		int id = obj.getId();
+
+		switch (id)
+		{
+			case DENSE_RUNESTONE_SOUTH_ID:
+				denseRunestoneSouth = obj;
+				break;
+			case DENSE_RUNESTONE_NORTH_ID:
+				denseRunestoneNorth = obj;
+				break;
+		}
+	}
+
+	@Subscribe
+	public void onGameObjectDespawned(GameObjectDespawned event)
+	{
+		switch (event.getGameObject().getId())
+		{
+			case DENSE_RUNESTONE_SOUTH_ID:
+				denseRunestoneSouth = null;
+				break;
+			case DENSE_RUNESTONE_NORTH_ID:
+				denseRunestoneNorth = null;
+				break;
+		}
+	}
+
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged event)
+	{
+		denseRunestoneSouthMineable = client.getVar(Varbits.DENSE_RUNESTONE_SOUTH_DEPLETED) == 0;
+		denseRunestoneNorthMineable = client.getVar(Varbits.DENSE_RUNESTONE_NORTH_DEPLETED) == 0;
+
+		updateDenseRunestoneState();
```

Comment:
This doesn't make any sense. Also you should probably just check the varbit state in the overlay.